tests/components/zha/test_sensor.py

```diff
@@ -779,26 +779,26 @@ async def test_unsupported_attributes_sensor(
         (
             1,
             1232000,
-            "123.20",
+            "123.2",
```

Comment:
What causes the difference here?

Reply:
Need to check this later again, but is a rounding thing, iirc

Reply:
This is the cast to float in line 613, I can avoid this and only do this if we really need to scale the value.

Reply:
Added this, we do the scaling now only when needed

Reply:
I've reverted that, I think we should change the `_formatter_function` to return either an `int` or a `float` rather than an `str`.
So we would still end up with `123.2` instead of `123.20`, but treating the values as a number instead of a string feels more correct, what do you think?

I would then create a second PR before merging this one.